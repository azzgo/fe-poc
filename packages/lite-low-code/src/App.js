import JsonEditor from "jsoneditor";
import AzzStorage from "azz-storage";
import React, { useEffect, useState } from "react";
import { Render } from "./render";
import { useRef } from "react";
import './widgets'

const jsonParser: IAzzStorageParser = {
  get(val, defaultVal) {
    try {
      const data = JSON.parse(val);
      return data.value;
    } catch (e) {
      return defaultVal;
    }
  },
  set(val) {
    return JSON.stringify({
      value: val,
    });
  },
};

const lcStore = new AzzStorage("lc-", jsonParser);

function App() {
  const jsoneditor = useRef(null);
  const [schema, setSchema] = useState(lcStore.get('schema', { type: "page" }));

  useEffect(() => {
    if (!jsoneditor.current) {
      jsoneditor.current = new JsonEditor(
        document.getElementById("jsoneditor"),
        { mode: "code", modes: ["code", "tree", "view"] }
      );
      jsoneditor.current.set(schema);
    }
  /* eslint-disable-next-line */
  }, []);

  const onSave = () => {
    const schema = jsoneditor.current.get();
    if (schema) {
      setSchema(jsoneditor.current.get());
      lcStore.set('schema', schema);
    }
  };

  const onReset = () => {
    setSchema({ type: "page" });
    jsoneditor.current.set({ type: "page" });
  };

  return (
    <div className="App row full-height">
      <div className="col-xs-6">
        <Render schema={schema} />
      </div>
      <div className="col-xs-6">
        <div className="row">
          <button onClick={onSave}>save</button>
          <button onClick={onReset}>reset</button>
        </div>
        <div id="jsoneditor" />
      </div>
    </div>
  );
}

export default App;
