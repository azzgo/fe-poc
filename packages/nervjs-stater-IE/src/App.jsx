
import React from 'react'
import ReactDOM from 'react-dom'

import { MasterLayout } from 'src/shared/components/MasterLayout/MasterLayout'

import 'src/shared/styles/flexboxgrid.min.css'
import 'src/shared/styles/global.css'
import 'src/shared/styles/icons.css'

class App extends React.Component {

  render () {
    return <div>hello</div>
  }
}

ReactDOM.render(
  <App />,
  document.getElementById('app'),
)
