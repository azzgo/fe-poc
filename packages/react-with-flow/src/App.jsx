// @flow
import notie from 'notie'
import React, { PureComponent, Fragment } from 'react'
import ReactDom from 'react-dom'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Redirect, Route } from 'react-router-dom'

import { store } from 'src/redux/store'

import { MasterLayout } from 'src/shared/components/MasterLayout/MasterLayout'

import LoginPage from 'src/app/Login/Login'
import 'src/shared/styles/flexboxgrid.min.css'
import 'src/shared/styles/global.css'
import 'src/shared/styles/icons.css'

notie.setOptions({
  alertTime: 2,
})

class App extends PureComponent<{}> {

  render () {
    return (
      <Router>
        <Fragment>
          <Route path="/login" component={LoginPage} />
          <Route path="/app" component={MasterLayout} />
          <Route path="/" exact render={() => <Redirect to="/app" />} />
        </Fragment>
      </Router>
    )
  }
}

ReactDom.render(
  <Provider store={store}>
    <App />
  </Provider>,
  (document.getElementById('app'): any),
)
