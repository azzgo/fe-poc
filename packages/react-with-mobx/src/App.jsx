import React, { useState } from "react";
import Home from "./pages/Home";
import About from "./pages/About";
import { counter } from "./store";
import {observer} from "mobx-react";

const App = (props) => {
  const [activeTab, setActiveTab] = useState("Home");
  
  return (
    <>
      <nav>
        <ul>
          <li>
            <a onClick={() => setActiveTab("Home")}>Home</a>
          </li>
          <li>
            <a onClick={() => setActiveTab("About")}>About</a>
          </li>
        </ul>
      </nav>
      <p>{counter.triggerWay}</p>
      <section>
        {activeTab === "Home" && <Home />}
        {activeTab === "About" && <About />}
      </section>
    </>
  );
};

export default observer(App);
