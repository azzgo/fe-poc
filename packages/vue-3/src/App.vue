<script>
import AppBar from "./components/Navbar.vue";
import { defineComponent } from "vue";

export default defineComponent({
  render() {
    return (
      <>
        <AppBar />
        <router-view />
      </>
    );
  },
});
</script>

