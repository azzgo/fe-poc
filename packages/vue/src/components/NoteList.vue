<template>
  <div class="row between-xs">
    <div class="row shadow-1 note-card col-xs-4" v-for="note in notes" :key="note.id">
      <div class="icon" @click="onCheckNote(note)">
        <i class="material-icons">check</i>
      </div>
      <div class="col-xs-12 title">{{note.title}}</div>
      <div class="col-xs-12 value">{{note.value}}</div>
    </div>
  </div>
</template>

<style lang="less" scoped>
.note-card {
  padding: 15px;
  padding-top: 35px;
  border-radius: 2px;
  width: 100%;
  position: relative;
}

.title {
  font-size: 1.2rem;
  font-weight: bold;
  text-align: left;
  color: rgba(0, 0, 0, 0.8);
}

.value {
  text-align: left;
  font-size: 1.4rem;
  font-weight: 200;
}

.icon {
  position: absolute;
  color: black;
  border: 1px solid lightgrey;
  background-color: white;
  font-size: 30px;
  top: -10px;
  left: -10px;
  width: 40px;
  height: 40px;
  border-radius: 100%;
  cursor: pointer;
}
</style>


<script>
export default {
  props: {
    notes: Array,
  },
  methods: {
    onCheckNote(note) {
      this.$emit('onCheckNote', note)
    }
  }
}
</script>
