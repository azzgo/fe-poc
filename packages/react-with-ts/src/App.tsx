import React, { PureComponent } from 'react';
import ReactDom from 'react-dom'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import { store } from 'src/store';
import { HomePage } from 'src/app/Home';
import { AboutPage } from 'src/app/About';

interface IProps {};

interface IState {};

class App extends PureComponent<IProps, IState> {
  public render(): JSX.Element {
    return (
      <Router>
        <React.Fragment>
          <Route path="/" exact component={HomePage} />
          <Route path="/about" component={AboutPage} />
        </React.Fragment>
      </Router>
    );
  }
}


ReactDom.render(
(
  <Provider store={store}>
    <App /> 
  </Provider>
)
, document.getElementById('app'))
