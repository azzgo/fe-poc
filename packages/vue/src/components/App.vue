<template>
  <div>
    <header class="app-bar row middle-xs">
      <a href="/" class="logo col-xs-10">
        Retain
      </a>
      <nav class="col-xs-2">
        <div class="row middle-xs between-xs">
          <a class="link" to="/app/about">About</a>
        </div>
      </nav>
    </header>
    <div class="warpper row center-xs">
      <div class="creator col-xs-6">
        <new-note-form @onCreateNewNote="createNote"></new-note-form>
      </div>
      <div class="notes col-xs-8">
        <note-list :notes="notes" @onCheckNote="checkNote"></note-list>
      </div>
    </div>
  </div>
</template>

<style type="text/less" scoped>
  .app-bar {
    height: 65px;
    padding: 5px 30px;
    background-color: #00BCD4;
  }
  .logo {
    color: white;
    font-size: 30px;
    font-weight: 300;
    cursor: pointer;
  }
  .link {
    color: white;
    font-size: 24px;
    font-weight: 400;
    cursor: pointer;
  }

  .warpper {
    margin-top: 50px;
  }

  .notes {
    padding-top: 50px;
  }

  .creator {
    margin-bottom: 40px;
  }
</style>

<script type="text/javascript">
import axios from 'axios'
import NewNoteForm from './NewNoteForm'
import NoteList from './NoteList'

export default {
  data() {
    return {
      notes: []
    }
  },
  mounted() {
    axios.post('http://localhost:3000/graphql', {
      query: `
        query {
          notes {
            id
            title
            value
          }
        }
      `
    }).then((res) => {
      this.notes = res.data.data.notes
    })
  },
  methods: {
    createNote(newNote) {
      axios.post('http://localhost:3000/graphql', {
        query: `
          mutation createNote($title: String!, $value: String!) {
            createNote(title: $title, value: $value) {
              id
              title
              value
            }
          }
        `,
        variables: {
          title: newNote.title,
          value: newNote.value,
        }
      }).then((res) => {
        this.notes = res.data.data.createNote
      })
    },
    checkNote({id}) {
      axios.post('http://localhost:3000/graphql', {
        query: `
          mutation checkNote($id: String!) {
            completeNote(id: $id) {
              id
              title
              value
            }
          }
        `,
        variables: {
          id,
        }
      }).then((res) => {
        this.notes = res.data.data.completeNote
      })
    },
  },
  components: {
    'new-note-form': NewNoteForm,
    'note-list': NoteList,
  }
}
</script>