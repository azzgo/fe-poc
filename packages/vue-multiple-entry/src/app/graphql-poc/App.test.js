import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import { shallowMount } from '@vue/test-utils'
import App from './App.vue'
import NoteList from './NoteList.vue'
import flushPromise from 'flush-promises'

// This sets the mock adapter on the default instance
const mock = new MockAdapter(axios)

test('App testing', async () => {
  mock.onPost('http://localhost:3000/graphql').reply(200, {
    data: {
      notes: [
        {
          id: '1',
          title: '今天吃什么',
          value: '豆花汤',
        },
        {
          id: '2',
          title: '今天喝什么',
          value: '白开水',
        },
      ],
    },
  })

  const comp = shallowMount(App, {
    stubs: {
      'note-list': NoteList,
    },
  })
  await flushPromise()
  expect(comp.html()).toMatchSnapshot()
})
