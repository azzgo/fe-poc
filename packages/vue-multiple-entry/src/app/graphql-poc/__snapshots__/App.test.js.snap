// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing 1`] = `
<div>
  <header class="app-bar row middle-xs">
    <a href="/" class="logo col-xs-10">
      Retain
    </a>
    <nav class="col-xs-2">
      <div class="row middle-xs between-xs">
        <a to="/app/about" class="link">About</a>
      </div>
    </nav>
  </header>
  <div class="warpper row center-xs">
    <div class="creator col-xs-6">
      <!---->
    </div>
    <div class="notes col-xs-8">
      <div class="row between-xs">
        <div class="row shadow-1 note-card col-xs-4">
          <div class="icon">
            <i class="material-icons">check</i>
          </div>
          <div class="col-xs-12 title">今天吃什么</div>
          <div class="col-xs-12 value">豆花汤</div>
        </div>
        <div class="row shadow-1 note-card col-xs-4">
          <div class="icon">
            <i class="material-icons">check</i>
          </div>
          <div class="col-xs-12 title">今天喝什么</div>
          <div class="col-xs-12 value">白开水</div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
